server/src/db.rs

```diff
@@ -389,7 +393,7 @@ impl Db {
         let fut = {
             let partition = self.partition(table_name, partition_key)?;
             let partition = LockableCatalogPartition {
-                db: self,
+                db: ArcDb(self),
```

Comment:
I think this would be much cleaner if it could be

```suggestion
                db: Arc::clone(self),
```